src/core/main/Reflection/TypeGetMethodRecursivelyExtensions.cs

```diff
@@ -17,11 +17,50 @@ public static class TypeGetMethodRecursivelyExtensions
         /// <param name="argTypes">The type of the arguments you would use</param>
         /// <returns>A <see cref="System.Reflection.MethodInfo" /></returns>
         /// <exception cref="System.MissingMethodException">Thrown if the method could not be found</exception>
+        /// <exception cref="System.Reflection.AmbiguousMatchException">Thrown if two or more methods match the method definition</exception>
         public static MethodInfo GetMethodRecursively(this Type type, string methodName, params Type[] argTypes)
         {
             var typeInfo = type.GetTypeInfo();
-            var method = typeInfo.GetMethod(methodName, argTypes);
+            MethodInfo method = null;
+            
+            // Get all the methods that satisfy the method declaration
+            var methods = typeInfo
+                .GetMethods()
+                .Where(m => m.Name == methodName && m.GetParameters().Length == argTypes.Length)
+                .ToList()
+                .FindAll(m =>
+                {
+                    // Make sure that each parameter satisfies the method definition
+                    for (var parameterIndex = 0; parameterIndex < argTypes.Length; parameterIndex++)
+                    {
+                        if (argTypes[parameterIndex] == null)
+                        {
+                            var parameter = m.GetParameters()[parameterIndex].ParameterType;
+                            if (parameter.GetTypeInfo().IsValueType && Nullable.GetUnderlyingType(parameter) == null)
```

Comment:
Could this be replaced by `if (parameter.IsNullable())`

Reply:
I don't believe this is the case as `parameter.IsNullable()` returns false for the string type.

Reply:
Ah yes, because string is this weird value type that can be set to null, but is not nullable.

Either we could "wrap" this in an extension method on `Type` like `Type.CanBeSetToNull`. This would make the code more semantic and thus easier to read.